frontend/src/utils/QuestionUtils.ts

```diff
@@ -98,7 +106,52 @@ export const getQuestionTexts = (question: QuestionElement[]): string[] => {
   return question
     .filter(
       (questionElement) =>
-        questionElement.type === QuestionElementType.QUESTION,
+        questionElement.type === QuestionElementType.QUESTION_TEXT,
     )
-    .map((questionElement) => questionElement.data as string);
+    .map(
+      (questionElement) =>
+        (questionElement.data as QuestionTextMetadata).questionText,
+    );
+};
+
+const formatMultipleChoiceRequest = (
+  data: MultipleChoiceData,
+): MultipleChoiceMetadata => {
+  return {
+    options: data.options.map((option) => option.value),
+    answerIndex: data.options.findIndex((option) => option.isCorrect),
+  };
+};
```

Comment:
for other question components, i just stored their `data` in their metadata types. but for multiple choice, since I needed an id field in its implementation, i just defined this extra helper function to convert it to its metadata type!